runtime/gc_modron_startup/mmparse.cpp

```diff
@@ -842,10 +910,16 @@ gcParseSovereignArguments(J9JavaVM *vm)
 	IDATA index = -1;
 	PORT_ACCESS_FROM_JAVAVM(vm);
 
-	if (!gcParseXlpOption(vm)) {
+	IDATA xlpOptionIndex = gcParseXlpOption(vm);
+	if (xlpOptionIndex == -1) {
 		goto _error;
 	}
 
+	/* Setup to handle additional objectheap arguments */
+	if (vm->largePageArgIndex > xlpOptionIndex && !gcProcessXXObjectHeapArguments(vm)) {
```

Comment:
please add extra brackets and do not rely on order of operations: `((vm->largePageArgIndex > xlpOptionIndex) && !gcProcessXXObjectHeapArguments(vm))`

Reply:
Thanks for catching that. 
See 
https://github.com/eclipse/openj9/pull/7476/commits/654eb0e91d7b27c0630b558f6423bd0b92ad7acf